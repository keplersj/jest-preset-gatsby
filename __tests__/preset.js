const path = require("path");

// Remove undeterministic data from test reports
expect.addSnapshotSerializer({
  print: (val, serialize) => {
    return serialize(path.relative(__filename, val));
  },
  test: val => typeof val === "string" && path.isAbsolute(val)
});

describe("Jest Preset for Gatsby Projects", () => {
  it("matches expectations", () => {
    const config = require("../jest-preset");
    expect(config).toMatchSnapshot();
  });
});
