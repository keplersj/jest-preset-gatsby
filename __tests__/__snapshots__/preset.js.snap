// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest Preset for Gatsby Projects matches expectations 1`] = `
Object {
  "globals": Object {
    "__PATH_PREFIX__": "",
  },
  "moduleFileExtensions": Array [
    "js",
    "jsx",
    "json",
    "node",
  ],
  "moduleNameMapper": Object {
    ".+\\\\.(css|styl|less|sass|scss)$": "identity-obj-proxy",
    ".+\\\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "jest-raw-loader",
  },
  "setupFiles": Array [
    "../../loadershim.js",
  ],
  "testPathIgnorePatterns": Array [
    "node_modules",
    ".cache",
  ],
  "testRegex": "(/__tests__/.*|(\\\\.|/)(test|spec))\\\\.(jsx?)$",
  "testURL": "http://localhost",
  "transform": Object {
    "^.+\\\\.jsx?$": "../../preprocess.js",
  },
  "transformIgnorePatterns": Array [
    "node_modules/(?!(gatsby)/)",
  ],
}
`;
